certbot/certbot/display/ops.py

```diff
@@ -19,58 +19,31 @@
 logger = logging.getLogger(__name__)
 
 
-def get_email(invalid: bool = False, optional: bool = True) -> str:
+def get_email(invalid: bool = False) -> Optional[str]:
     """Prompt for valid email address.
 
     :param bool invalid: True if an invalid address was provided by the user
-    :param bool optional: True if the user can use
-        --register-unsafely-without-email to avoid providing an e-mail
 
     :returns: e-mail address
     :rtype: str
 
     :raises errors.Error: if the user cancels
 
     """
-    invalid_prefix = "There seem to be problems with that address. "
-    msg = "Enter email address (used for urgent renewal and security notices)\n"
-    unsafe_suggestion = ("\n\nIf you really want to skip this, you can run "
-                         "the client with --register-unsafely-without-email "
-                         "but you will then be unable to receive notice about "
-                         "impending expiration or revocation of your "
-                         "certificates or problems with your Certbot "
-                         "installation that will lead to failure to renew.\n\n")
-    if optional:
-        if invalid:
-            msg += unsafe_suggestion
-            suggest_unsafe = False
-        else:
-            suggest_unsafe = True
-    else:
-        suggest_unsafe = False
+    msg = "Enter email address or hit Enter to skip.\n"
+    if invalid:
+        msg = "The server reported a problem with your email address. " + msg
 
     while True:
-        try:
-            code, email = display_util.input_text(invalid_prefix + msg if invalid else msg,
-                                                  force_interactive=True)
-        except errors.MissingCommandlineFlag:
-            msg = ("You should register before running non-interactively, "
-                   "or provide --agree-tos and --email <email_address> flags.")
-            raise errors.MissingCommandlineFlag(msg)
+        code, email = display_util.input_text(msg, force_interactive=True)
```

Comment:
If we're using a noninteractive display (in noninteractive mode or with `--quiet`), since there's no default passed in here, `NoninteractiveDisplay._interaction_fail` will raise `errors.MissingCommandlineFlag`.

Otherwise, `FileDisplay._return_default` will call `FileDisplay._can_interact`. Since we've set `force_interactive` here, `_can_interact` will always be true, so `_return_default` will always be `None`, and we'll always (at least attempt to) show the prompt. 

I suspect what you actually want here is to set a default of an empty string, and remove `force_interactive`. Then we'll just do a best-efforts attempt to be interactive, and if not, just return an empty string, possibly printing a message about skipping an interaction.

Reply:
Also you probably do want to add back in the `invalid_prefix` bit; that means that last time we hit an unsafe email and we're trying again. If you really don't, then might as well delete the `invalid = True` line but I do think we want that.